packages/nhsuk-frontend/src/nhsuk/components/file-upload/file-upload.mjs

```diff
@@ -0,0 +1,492 @@
+import { ElementError } from '../../errors/index.mjs'
+import { I18n } from '../../i18n.mjs'
+
+/**
+ * File upload component
+ *
+ * @preserve
+ * @augments ConfigurableComponent<FileUploadConfig>
+ */
+export class FileUpload extends ConfigurableComponent {
+  /**
+   * @private
+   * @type {HTMLFileInputElement}
+   */
+  $input
+
+  /**
+   * @private
+   */
+  $button
+
+  /**
+   * @private
+   */
+  $status
+
+  /** @private */
+  i18n
+
+  /** @private */
+  id
+
+  /** @private */
+  $announcements
+
+  /**
+   * @private
+   * @type {boolean | undefined}
+   */
+  enteredAnotherElement
+
+  /**
+   * @param {Element | null} $root - File input element
+   * @param {FileUploadConfig} [config] - File Upload config
+   */
+  constructor($root, config = {}) {
+    super($root, config)
+
+    const $input = this.$root.querySelector('input')
+
+    if ($input === null) {
+      throw new ElementError({
+        component: FileUpload,
+        identifier: 'File inputs (`<input type="file">`)'
+      })
+    }
+
+    if ($input.type !== 'file') {
+      throw new ElementError({
+        component: FileUpload,
+        identifier:
+          'File input (`<input type="file">`) attribute (`type`) is not `file`'
+      })
+    }
+
+    this.$input = /** @type {HTMLFileInputElement} */ ($input)
+
+    if (!this.$input.id) {
+      throw new ElementError({
+        component: FileUpload,
+        identifier: 'File input (`<input type="file">`) attribute (`id`)'
+      })
+    }
+
+    this.id = this.$input.id
+
+    this.i18n = new I18n(this.config.i18n, {
+      // Read the fallback if necessary rather than have it set in the defaults
+      locale: closestAttributeValue(this.$root, 'lang')
+    })
+
+    const $label = this.findLabel()
+    // Add an ID to the label if it doesn't have one already
+    // so it can be referenced by `aria-labelledby`
+    if (!$label.id) {
+      $label.id = `${this.id}-label`
+    }
+
+    // We need to copy the 'id' of the root element
+    // to the new button replacement element
+    // so that focus will work in the error summary
+    this.$input.id = `${this.id}-input`
+
+    // Hide the native input
+    this.$input.setAttribute('hidden', 'true')
+
+    // Create the file selection button
+    const $button = document.createElement('button')
+    $button.classList.add('nhsuk-file-upload-button')
+    $button.type = 'button'
+    $button.id = this.id
+    $button.classList.add('nhsuk-file-upload-button--empty')
+
+    // Copy `aria-describedby` if present so hints and errors
+    // are associated to the `<button>`
+    const ariaDescribedBy = this.$input.getAttribute('aria-describedby')
+    if (ariaDescribedBy) {
+      $button.setAttribute('aria-describedby', ariaDescribedBy)
+    }
+
+    // Create status element that shows what/how many files are selected
+    const $status = document.createElement('span')
+    $status.className = 'nhsuk-body-m nhsuk-file-upload-button__status'
```

Comment:
Noticed we added `-m` here but aren't `nhsuk-body-m` and `nhsuk-body` essentially the same?

Reply:
I looked, and there is no `nhsuk-body` class in out CSS! It’s just that a bare `p` element will get the default `nhsuk-body-m` styles applied. Probably something to fix in another PR?

Reply:
Updated to use `nhsuk-body`.